components/brave_rewards/core/wallet_provider/solana/solana_wallet_provider.cc

```diff
@@ -0,0 +1,198 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/core/wallet_provider/solana/solana_wallet_provider.h"
+
+#include <cmath>
+#include <utility>
+#include <vector>
+
+#include "base/base64url.h"
+#include "base/strings/strcat.h"
+#include "base/strings/string_util.h"
+#include "brave/components/brave_rewards/core/common/signer.h"
+#include "brave/components/brave_rewards/core/endpoint/rewards/rewards_util.h"
+#include "brave/components/brave_rewards/core/global_constants.h"
+#include "brave/components/brave_rewards/core/logging/event_log_keys.h"
+#include "brave/components/brave_rewards/core/state/state_keys.h"
+#include "brave/components/brave_rewards/core/wallet/wallet_util.h"
+#include "brave/components/brave_rewards/core/wallet_provider/linkage_checker.h"
+#include "net/base/url_util.h"
+#include "url/gurl.h"
+
+namespace brave_rewards::internal {
+
+namespace {
+
+constexpr char kSplBatTokenMint[] =
+    "EPeUFDgHRxs9xxEPVaL6kfGQvCon7jmAWKVUHuux1Tpz";
+constexpr base::TimeDelta kPollingInterval = base::Seconds(10);
```

Comment:
Did we get an issue open to refactor this away from the polling logic later? I remember we had talked about it, but opted to stick with this for now.

Reply:
I don't think we have an issue yet - I'll defer to @Miyayes on this one.

Reply:
Filed https://github.com/brave/brave-browser/issues/35370